up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,613 @@ SPDX-FileType: DOCUMENTATION
+--
+
+[.specitem,oft-sid="req~usubscription-reset~1",oft-needs="impl,utest"]
+--
+When the `Reset()` function is called, the uSubscription service *MUST* flush all stored subscription information, including any persistently stored subscriptions.
+--
+
+[.specitem,oft-sid="req~usubscription-reset-only-usubscription~1",oft-needs="impl,utest"]
+--
+When receiving a `Reset()` call from a source that is not another uSubscription services (i.e. from source URIs where uEntity ID (`ue_id`) does not equal _0x0_), uSubscription service *MUST* return a failure status message with link:../../../up-core-api/uprotocol/v1/ucode.proto[`UCode`] `PERMISSION_DENIED`.
+--
 
 == uSubscription Sequences
 
 In the following section, we will elaborate on the various subscription flows for local and remote topics. When a consumer subscribes to a remote topic, it is the responsibility of the (local) uSubscription service to relay the subscription request to the remote uSubscription service as can be seen in the sequence diagrams below.
 
 NOTE: Throughout this section we will use the sample topic `//Device1/uexample/1/resource#Event` to illustrate the various sequences. The above-mentioned topic will be replaced with `_topic_` in the diagrams
 
-
-
 === Subscription
 
 Subscription flow will show how a subscriber can subscribe to the example topic when uApp is on the same device (local subscriptions) or remote device (remote subscriptions).
 
 ==== Within a uDevice
 
 .Local Subscription Flow
-image::local_subscribe.svg[Local Subscription Flow]
+[mermaid]
+ifdef::env-github[[source,mermaid]]
+----
+sequenceDiagram
+    box White Device1
+        actor uApp
+        participant uSubscription
+        participant SubscriptionChange
+    end
+
+    uApp->>+uSubscription: Subscribe(SubscriptionRequest)   
+
+    alt success
+        uSubscription-->>uApp: SubscriptionResponse(SUBSCRIBED)
+        uSubscription--)SubscriptionChange: Update(SUBSCRIBED)
+    else failure
+        uSubscription-->>-uApp: SubscriptionResponse(UNSUBSCRIBED)
+    end
+----
 
 ==== Between uDevices
 
 .Remote Subscription Flow
-image::remote_subscription.svg[Remote Subscription Flow]
+[mermaid]
+ifdef::env-github[[source,mermaid]]
+----
+sequenceDiagram
+    box White Device1
+        actor uApp
+        participant local uSubscription
+    end
+    box White Device2
+        participant remote uSubscription
+        participant uEntity
+    end
+    
+    uEntity->>+remote uSubscription: RegisterForNotification()
+    uApp->>+local uSubscription: Subscribe(SubscriptionRequest)
+
+    alt first subscription
+        local uSubscription-->>uApp: SubscriptionResponse(SUBSCRIPTION_PENDING)
+        local uSubscription-->>remote uSubscription: Subscribe(SubscriptionRequest)
+        alt success
+            remote uSubscription-->>local uSubscription: SubscriptionResponse(SUBSCRIBED)
```

Comment:
We do not send SubscriptionResponse twice to the client as the RPC pattern is unary (and not a flow). When the local usubscriptionService receives the SubscriptionResponse(SUBSCRIBED), it will send the Update notification to the local subscriber